src/components/Topology/topology-breadcrumbs.js

```diff
@@ -1,62 +1,118 @@
 
-function isTopologyEmpty(topology) {
-  return (
-    topology == null ||
-    (isEmpty(topology.id) && topology.components.length === 0)
-  );
-}
+export function TopologyBreadcrumbs({ clusterId, groupId, podOrNodeId }) {
+  const [cluster, setCluster] = useState();
+  const [group, setGroup] = useState();
+  const [podOrNode, setPodOrNode] = useState();
+  const { loading, setLoading } = useLoader();
 
-export function TopologyBreadcrumbs({
-  topology,
-  topologyId
-  // depth
-}) {
-  const [_topology, setTopology] = useState(topology);
+  async function loadClusterDetails() {
+    if (!clusterId) {
+      return;
+    }
+    setLoading(true);
+    try {
+      const results = await getTopology({ id: clusterId });
+      const value = results.data.find((item) => item.id === clusterId);
+      setCluster(value);
+    } catch (ex) {
+      setCluster(null);
+    }
+    setLoading(false);
+  }
 
-  useEffect(() => {
-    if (isEmpty(topologyId) && isTopologyEmpty(_topology)) {
+  async function loadGroupDetails() {
+    if (!groupId) {
       return;
     }
-    if (topologyId != null && _topology == null) {
-      getTopology({ id: topologyId }).then((results) => {
-        setTopology(results.data[0]);
-      });
+    setLoading(true);
+    try {
+      const results = await getTopology({ id: groupId });
+      const value = results.data.find((item) => item.id === groupId);
+      setGroup(value);
+    } catch (ex) {
+      setGroup(null);
     }
-  }, [_topology, topologyId]);
+    setLoading(false);
+  }
 
-  if (_topology == null && isEmpty(topologyId)) {
-    return "";
+  async function loadPodOrNodeDetails() {
+    if (!podOrNodeId) {
+      return;
+    }
+    setLoading(true);
+    try {
+      const results = await getTopology({ id: podOrNodeId });
+      const value = results.data.find((item) => item.id === podOrNodeId);
+      setPodOrNode(value);
+    } catch (ex) {
+      setPodOrNode(null);
+    }
+    setLoading(false);
   }
-  if (_topology == null) {
+
+  useEffect(() => {
+    loadClusterDetails();
+  }, [clusterId]);
+
+  useEffect(() => {
+    loadGroupDetails();
+  }, [groupId]);
+
+  useEffect(() => {
+    loadPodOrNodeDetails();
+  }, [podOrNodeId]);
+
+  if (loading) {
     return <Loading text=".." />;
   }
 
   return (
     <>
-      {/* {depth > 0 &&
-        _topology.parent_id != null &&
-        _topology.parent_id != _topology.id && (
-          <>
-            <TopologyBreadcrumbs
-              topologyId={_topology.parent_id}
-              depth={depth - 1}
-            />
-          </>
-        )} */}
-      &nbsp;/&nbsp;
-      <Link
-        to={`/topology/${_topology.id}`}
-        className="flex flex-nowrap hover:text-gray-500 my-auto "
-      >
-        <Icon name={_topology.icon} size="xl" className="mr-1" />
-        {_topology.name || _topology.title}
+      <Link to="/topology" className="hover:text-gray-500 ">
+        Topology
       </Link>
+      {cluster && (
```

Comment:
The breadcrumbs can't be hardcoded like this, as the hierarchy is arbritrary

Reply:
I went with approach because of following reason.
- Currently we are not returning proper parent_id when we fetch topology by id
   `curl 'http://localhost:3004/canary/api/topology?id=018138c7-c631-ecd2-6b0c-cf6559ebebc2' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Cookie: _ga=GA1.1.27620775.1611673700; amplitude_id_eb9e4621db21faf713ee50c637f3cf36=eyJkZXZpY2VJZCI6IjcxMzgyYzZiLTYyMjEtNDMxNi1iY2RiLTRkMWQ0OGQyOGY2ZFIiLCJ1c2VySWQiOm51bGwsIm9wdE91dCI6ZmFsc2UsInNlc3Npb25JZCI6MTYxMjAyNTc0MTE4MCwibGFzdEV2ZW50VGltZSI6MTYxMjAyNTg2MzI0MiwiZXZlbnRJZCI6MCwiaWRlbnRpZnlJZCI6MCwic2VxdWVuY2VOdW1iZXIiOjB9; _ga_M4NQSZJKMF=GS1.1.1613889114.8.1.1613891920.0; amplitude_id_7f4b058e1a8c4a8746473d8f92d1497f=eyJkZXZpY2VJZCI6ImEzZmVhNTVmLTdjNTMtNGM1Yy04N2RjLTQxOWU5MjYxMDFkY1IiLCJ1c2VySWQiOiIyMTgiLCJvcHRPdXQiOmZhbHNlLCJzZXNzaW9uSWQiOjE2MzI5MjM0OTgzNjIsImxhc3RFdmVudFRpbWUiOjE2MzI5MjQzNDkwMjAsImV2ZW50SWQiOjMzMywiaWRlbnRpZnlJZCI6NDQ5LCJzZXF1ZW5jZU51bWJlciI6NzgyfQ==; _hjid=e312216c-e783-456f-a540-c29c0546adbb; _hjSessionUser_2207626=eyJpZCI6IjRlOTliMDM3LTE1Y2EtNTZjNC1iMDE3LWE2NmI4YjQ2OWM2YSIsImNyZWF0ZWQiOjE2NDMzODc3OTA5NTUsImV4aXN0aW5nIjp0cnVlfQ==; __gads=ID=b7153cc4ae6e4c0b-22bd4e49e1d00070:T=1646676290:RT=1646676290:S=ALNI_MYQhzsuypq1E4Ni7mB1sVrdHFdwVQ; _ga_94LF6Z0YYR=GS1.1.1650807302.204.0.1650807302.0' \
  -H 'DNT: 1' \
  -H 'Pragma: no-cache' \
  -H 'Prefer: return=representation' \
  -H 'Referer: http://localhost:3004/topology/018138c7-c627-98b8-65e2-c9fd9fb6e9db/018138c7-c631-ecd2-6b0c-cf6559ebebc2' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="102", "Google Chrome";v="102"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --compressed`

Is there any limit on topology depth  ?